lib/subscription/subscription-store.ts

```diff
@@ -1,72 +1,47 @@
 /// <reference path='../../typings/tsd.d.ts' />
 
 import _ = require('lodash');
-import Subscription = require('./subscription');
+import Subscription = require('./subscription-buffer');
+import Map = require('../util/map');
+import interfaces = require('../interface');
 
 export = SubscriptionStore;
 
-class SubscriptionStore<T extends Subscription> {
+class SubscriptionStore extends Map<Subscription> {
```

Comment:
I generally agree that I usually think of it as being a "has a" relationship, not "is a". That is, I think of SubscriptionStore as having a map that it uses to store its internal state, rather than being a map. The main downside is that you'll need to define a bunch of functions to just delegate to the map member, though.